src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java

```diff
@@ -35,4 +37,24 @@ public void testUnSortedRoundTrip() throws Exception {
         SamAssertionUtils.assertSamsEqual(outBam, inBam);
     }
 
+    /**
+     * Test GCS access using Spark NIO.
+     *
+     * For this to work, the settings in src/main/resources/core-site.xml must be correct,
+     * and the project name and credential file it points to must be present.
+     */
+    @Test(dataProvider = "gcsTestingData", groups = "bucket")
+    public void testGCSInputsAndOutputsWithSparkNio(final String gcsInput, final String outputExtension,
+                                        final boolean outputToGCS, final File expectedOutput) {
+        final String gcsInputPath = getGCPTestInputPath() + gcsInput;
+        final String outputPrefix = outputToGCS ? getGCPTestStaging() : "testGCSInputsAndOutputs";
+        final String outputPath = BucketUtils.getTempFilePath(outputPrefix, outputExtension);
+
+        final ArgumentsBuilder argBuilder = new ArgumentsBuilder();
+        argBuilder.addArgument("input", gcsInputPath)
+                .addArgument("output", outputPath)
+                .addBooleanArgument(GATKSparkTool.USE_NIO, true);
+        runCommandLine(argBuilder);
```

Comment:
This test is probably sufficient but really most of our tests should at least assert something specific. To that end I would think you probably want to make this test asserting equality of the output.

Reply:
This test is a copy of the one from the non-Spark test. I think the idea is to check that it doesn't blow up.

Reply:
Done